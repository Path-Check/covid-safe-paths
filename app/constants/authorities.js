export const PUBLIC_DATA_URL =
  'https://raw.githubusercontent.com/beoutbreakprepared/nCoV2019/master/latest_data/latestdata.csv';

export const AUTHORITIES_LIST_URL =
  'https://raw.githubusercontent.com/tripleblindmarket/safe-places/develop/healthcare-authorities.yaml';
