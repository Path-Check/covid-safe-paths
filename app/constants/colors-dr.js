const colors = {
  BLUE_RIBBON: '#0161F2',
  LIGHT_GRAY: '#EFEFEF',
  LIGHT_BLUE: '#EFF4F9',
  DOVE_GRAY: '#6B6B6B',
  SUN: '#FEB313',
  PINK: '#F1485B',
  PURPLE: '#525BD9',
  CONFETTI: '#EAEB5E',
  YUKON_GOLD: '#666804',
  GREEN: '#7ED321',
};

export default colors;
