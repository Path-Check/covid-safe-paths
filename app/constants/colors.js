const colors = {
  BLACK: '#000',
  WHITE: '#FFF',
  DODGER_BLUE: '#428AF8',
  SILVER: '#BEBEBE',
  TORCH_RED: '#F8262F',
  MISCHKA: '#E5E4E6',
  APP_BACKGROUND: '#FFF8ED',
  PRIMARY_TEXT: '#000',
  GREEN: '#32A852',
  VIOLET: '#6C3794',

  // For notifications
  HIGHEST_RISK: '#e74c3c',
  MIDDLE_RISK: '#e67e22',
  LOWER_RISK: '#FFFF33',
  LOWEST_RISK: 'brown',

  REG_BUTTON: '#428AF8',
  POS_BUTTON: '#32A852',
  NEG_BUTTON: '#F8262F',
  SENS_BUTTON: '#6C3794',

  BLUE_BUTTON: '#05369B',
  BLUE_TO_BUTTON: '#185BD3',
  RED_BUTTON: '#E3554E',
  RED_TO_BUTTON: '#E97C79',
  GRAY_BUTTON: '#868686',
  GRAY_TO_BUTTON: '#868686',

  BLUE_LINK: '#007AFF',
};

export default colors;
