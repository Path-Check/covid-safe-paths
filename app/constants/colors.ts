const colors = {
  BLACK: '#000',
  WHITE: '#FFF',
  DODGER_BLUE: '#428AF8',

  PRIMARY_50: '#6979F8',
  SECONDARY_10: '#F8F8FF',
  SECONDARY_50: '#E5E7FA',
  SECONDARY_100: '#A5AFFB',

  SILVER: '#BEBEBE',
  TORCH_RED: '#F8262F',
  MISCHKA: '#E5E4E6',
  PRIMARY_TEXT: '#000',
  SUCCESS: '#41dca4', // Green
  WARNING: '#ffc000', // Orange
  VIOLET_ALPHA_06: '#f7f8ff', // cannot use transparency
  VIOLET: '#4051DB',
  TRANSPARENT: 'transparent',
  DARK_ALPHA: 'rgba(0, 0, 0, 0.7)',

  //Onboarding
  ONBOARDING_ICON_LIGHT_BLUE: '#D3D7F8',
  ONBOARDING_ICON_LIGHT_YELLOW: '#F9EDCC',

  // For notifications
  HIGHEST_RISK: '#e74c3c',
  MIDDLE_RISK: '#e74c3c',
  LOWER_RISK: '#e74c3c',
  LOWEST_RISK: '#e74c3c',

  REG_BUTTON: '#428AF8',
  POS_BUTTON: '#32A852',
  NEG_BUTTON: '#F8262F',
  SENS_BUTTON: '#6C3794',
  VIOLET_BUTTON: '#6979F8',
  VIOLET_BUTTON_DARK: '#3A4CD7',
  VIOLET_BUTTON_LIGHT: '#E5E7FA',

  VIOLET_TEXT: '#3C4ED8',
  VIOLET_TEXT_DARK: '#1F2C9B',
  VIOLET_TEXT_LIGHT: '#4754C5',

  GRAY_BACKGROUND: '#DADADA',

  INTRO_WHITE_BG: '#F7F8FF',
  PULSE_WHITE: '#FFFFFF40',

  DIVIDER: '#A5AFFB',

  BLUE_BUTTON: '#05369B',
  BLUE_TO_BUTTON: '#185BD3',
  RED_BUTTON: '#E3554E',
  RED_TO_BUTTON: '#E97C79',
  GRAY_BUTTON: '#868686',
  GRAY_TO_BUTTON: '#868686',

  MAP_LINE_STROKE: '#4052DB',

  BOTTOM_SHEET_HEADER_INDICATOR: '#E1E4FF',

  BLUE_LINK: '#007AFF',

  RED_TEXT: '#FF5656',

  NAV_BAR_VIOLET: '#5061E6',
  NAV_BAR_BORDER: 'rgba(189, 195, 199,0.6)',

  DARK_GRAY: '#3C475B',
  LIGHT_GRAY: '#EEEEEE',

  MONO_DARK: '#202020',
  MONO_SECONDARY: '#757677',

  LINK: '#FFDC58',

  UNDERLAY: '#E5E7FA',
};

export default colors;
