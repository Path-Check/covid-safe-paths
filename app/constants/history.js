/**
 * Max esposure reporting window in days
 */
export const MAX_EXPOSURE_WINDOW = 14;

/**
 * The value in minutes of each "bin" in the crossed path data.
 */
export const BIN_DURATION = 5;

/**
 * Format of a single history item
 *
 * @typedef {{
 *   date: import("dayjs").Dayjs,
 *   exposureMinutes: number,
 * }} HistoryDay
 */

/**
 * Exposure history
 *
 * @typedef {!HistoryDay[]} History
 */
