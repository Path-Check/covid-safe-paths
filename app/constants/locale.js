export const CONVERT_ISO_TO_IETF = {
  es_419: 'es-us',
  tl: 'tl-ph',
  zh_Hant: 'zh',
};
