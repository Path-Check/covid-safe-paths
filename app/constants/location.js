/**
 * The max amount of attempts to get desired accuracy.
 * @type {number}
 */
export const MAX_ACCURACY_RETRY = 2;

/**
 * The desired accuracy (in meters) quality which serves as a threshold.
 *
 * @type {number}
 */
export const DESIRED_ACCURACY_QUALITY = 30;

/**
 * The desired location interval, and the minimum acceptable interval.
 * Time (in milliseconds) between location information polls.
 * E.g. 60000*5 = 5 minutes
 * @type {number}
 */
export const LOCATION_INTERVAL = 60000 * 5;

/**
 * Maximum time that we will backfill for missing data
 * Time (in milliseconds).  60000 * 60 * 8 = 24 hours
 * @type {number}
 */
export const MAX_BACKFILL_TIME = 60000 * 60 * 24;
