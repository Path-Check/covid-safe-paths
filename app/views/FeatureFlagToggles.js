/* eslint-disable react-native/no-raw-text */
import React, { useEffect, useCallback, useRef } from 'react';
import { StyleSheet, View, FlatList } from 'react-native';

import {
  NavigationBarWrapper,
  Switch,
  Typography,
  Button,
} from '../components';
import { useDispatch, useSelector } from 'react-redux';
import toggleFeatureFlagAction from '../store/actions/featureFlags/toggleFeatureFlagAction';
import { Spacing } from '../styles';
import { FeatureFlagOption } from '../store/types';
import { initDevLanguages, initProdLanguages } from '../locales/languages';
import toggleAllowFeatureFlagsAction from '../store/actions/featureFlags/toggleAllowFeatureFlagsEnabledAction';

const flagToName = (flag) => {
  switch (flag) {
    case FeatureFlagOption.CUSTOM_URL:
      return 'Custom Yaml URL';
    case FeatureFlagOption.DOWNLOAD_LOCALLY:
      return 'Download Locally';
    case FeatureFlagOption.GOOGLE_IMPORT:
      return 'Import from Google';
    case FeatureFlagOption.DEV_LANGUAGES:
      return 'All Language Options';
    // For development ease:
    default:
      return flag;
  }
};

export const FlagToggleRow = ({ flag }) => {
  const dispatch = useDispatch();
  const flags = useSelector((state) => state.featureFlags.flags);
  const isActive = flags[flag];
  const toggleActive = () =>
    dispatch(toggleFeatureFlagAction({ flag, overrideValue: !isActive }));
  return (
    <View style={styles.toggleRow}>
      <Switch value={isActive} onValueChange={toggleActive} />
      <View style={styles.toggleRowText}>
        <Typography use='body1' bold>
          {flagToName(flag)}
        </Typography>
      </View>
    </View>
  );
};

export const FeatureFlagsScreen = ({ navigation }) => {
  const flagMap = useSelector((state) => state.featureFlags.flags);
  const flags = Object.keys(flagMap);
  const devLanguagesEnabled = flagMap[FeatureFlagOption.DEV_LANGUAGES];
  const isLoaded = useRef(true);

  // Only runs when language flag is toggled
  const toggleLanguages = useCallback(() => {
    if (devLanguagesEnabled) {
      initDevLanguages();
    } else {
      initProdLanguages();
    }
  }, [devLanguagesEnabled]);

  // Dev languages requires an init step, not just conditional render
  useEffect(() => {
    if (isLoaded.current) {
      isLoaded.current = false;
      return;
    } else {
      toggleLanguages();
    }
  }, [toggleLanguages]);
  const dispatch = useDispatch();

  const disableFeatureFlags = () => {
    dispatch(toggleAllowFeatureFlagsAction({ overrideValue: false }));
    navigation.goBack();
  };

  return (
    <NavigationBarWrapper
      title={'Feature Flags'}
      onBackPress={() => navigation.goBack()}>
      <View style={{ padding: Spacing.large, flex: 1 }}>
        <Typography use='body1' style={styles.headerText} bold>
          Notice
        </Typography>
        <Typography use='body2' style={styles.headerText}>
          This screen is for enabling advanced debugging features
        </Typography>
        <View style={{ height: Spacing.large }} />
        <FlatList
          alwaysBounceVertical={false}
          data={flags}
          keyExtractor={(_, i) => `${i}`}
          renderItem={({ item: flag }) => <FlagToggleRow flag={flag} />}
        />
        <Button label='Disable Feature Flags' onPress={disableFeatureFlags} />
      </View>
    </NavigationBarWrapper>
  );
};

const styles = StyleSheet.create({
  headerText: {
    textAlign: 'center',
    paddingTop: 7.5,
    marginHorizontal: 15,
  },
  toggleRow: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 10,
    paddingHorizontal: 5,
    justifyContent: 'flex-start',
  },
  toggleRowText: {
    paddingLeft: 15,
  },
});
