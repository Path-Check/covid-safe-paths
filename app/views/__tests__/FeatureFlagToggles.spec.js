import { act, fireEvent, render, wait } from '@testing-library/react-native';
import React from 'react';

import * as flagsEnv from '../../constants/flagsEnv';
import * as Flags from '../../helpers/Flags';
import { FeatureFlagsScreen } from '../FeatureFlagToggles';

const { FlagsProvider } = Flags;
const buildTimeFlags = { feature1: true };

flagsEnv.buildTimeFlags = buildTimeFlags;

jest.mock('../../helpers/General', () => {
  return {
    GetStoreData: jest.fn().mockResolvedValue(false),
  };
});

it('renders default values from the flags provider', async () => {
  const { asJSON } = render(
    <FlagsProvider>
      <FeatureFlagsScreen />
    </FlagsProvider>,
  );

  expect(asJSON()).toMatchSnapshot();
});

// TODO
it.skip('renders toggle events', async () => {
  const { getByTestId } = render(
    <FlagsProvider>
      <FeatureFlagsScreen />
    </FlagsProvider>,
  );

  const toggle = getByTestId('feature1');

  await act(() => {
    fireEvent.valueChange(toggle, false);
  });

  await wait();

  expect(toggle.props.value).toBe(false);
});
