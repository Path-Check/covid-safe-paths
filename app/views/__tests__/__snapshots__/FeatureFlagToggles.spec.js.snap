// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default values from the flags provider 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#1f2c9b",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1f2c9b",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <SvgXml
            xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontSize": 17,
                "lineHeight": 28,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 19,
                "fontWeight": "500",
                "lineHeight": 40,
              },
            ]
          }
        >
          Feature Flags
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f8f8f8",
          "height": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontSize": 17,
              "lineHeight": 28,
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "marginHorizontal": 15,
              "paddingTop": 7.5,
              "textAlign": "center",
            },
          ]
        }
      >
        Notice
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#4e4e4e",
              "fontSize": 17,
              "lineHeight": 28,
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "marginHorizontal": 15,
              "paddingTop": 7.5,
              "textAlign": "center",
            },
          ]
        }
      >
        This screen is only available when running the application in development or testing
      </Text>
      <View
        style={
          Object {
            "marginHorizontal": 10,
            "paddingTop": 20,
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="button"
            onTintColor="#ff5656"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            testID="switch"
            thumbTintColor="#ffffff"
            tintColor="#606060"
            value={false}
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#2e2e2e",
                  "fontSize": 17,
                  "lineHeight": 28,
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "paddingLeft": 15,
                },
              ]
            }
          >
            Exposure Mode
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="button"
            onTintColor="#41dca4"
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            testID="FOO_BAR"
            thumbTintColor="#ffffff"
            tintColor="#4e4e4e"
            value={true}
          />
          <View
            style={
              Object {
                "paddingLeft": 15,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              FOO BAR
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4e4e4e",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Default:
              <Text
                style={
                  Array [
                    Object {
                      "color": "#6979f8",
                      "fontSize": 15,
                      "lineHeight": 24,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "fontStyle": "italic",
                    },
                  ]
                }
              >
                 On
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
